### ISSUE TITLE
[Release] 0.14

### ISSUE THREAD

**Issue by jhamman** [pydata/xarray#3380]
#3358 is going to make some fairly major changes to the minimum supported versions of required and optional dependencies. We also have a few bug fixes that have landed since releasing 0.13 that would be good to get out.

From what I can tell, the following pending PRs are close enough to get into this release. 
- [ ] ~tests for arrays with units #3238~
- [x] map_blocks #3276
- [x] Rolling minimum dependency versions policy #3358
- [x] Remove all OrderedDict's (#3389)
- [x] Speed up isel and \_\_getitem\_\_ #3375
- [x] Fix concat bug when concatenating unlabeled dimensions. #3362
- [ ] ~Add hypothesis test for netCDF4 roundtrip #3283~
- [x] Fix groupby reduce for dataarray #3338 
- [x] Need a fix for https://github.com/pydata/xarray/issues/3377

Am I missing anything else that needs to get in?

I think we should aim to wrap this release up soon (this week). I can volunteer to go through the release steps once we're ready.

**Comment by dcherian** [comment]
Added #3338

**Comment by crusaderky** [comment]
Nuke away all OrderedDicts! (blocked by #3358)

**Comment by dcherian** [comment]
This looks like it may have fallen out of the indexing refactor; definitely should be fixed: https://github.com/pydata/xarray/issues/3377

**Comment by keewis** [comment]
I don't know if I will have the time to finish #3238 this week, even if I limit it to only testing `DataArray`.

**Comment by jhamman** [comment]
> Nuke away all OrderedDicts! (blocked by #3358)

@crusaderky - have you started on this yet?

**Comment by crusaderky** [comment]
@jhamman no - I should have it ready by the end of the weekend.

**Comment by jhamman** [comment]
> @jhamman no - I should have it ready by the end of the weekend.

I plan to make the release on Friday. I can do this task before then unless you really want to do it :wink:.

**Comment by shoyer** [comment]
Should we also make `assign` and `assign_coords` no longer sort their arguments? We can do that now that dictionary are always ordered

**Comment by crusaderky** [comment]
@jhamman never release to production on a Friday! 🏖 
Jokes aside, I'm happy to leave the burden to you

**Comment by jhamman** [comment]
See #3389 for the PR removing OrderedDicts.

**Comment by jhamman** [comment]
>  Need a fix for #3377

See #3393

**Comment by jhamman** [comment]
I think we're feature complete here. There are three PRs that are close (#3238, #3283, #3285) but probably don't need to hold up a release. I have some time Monday morning to issue the release so we'll go with that. Nice work all!

**Comment by dcherian** [comment]
#3364 could be useful for the hackathon but definitely not necessary.

Thanks for pushing this along Joe.

**Comment by crusaderky** [comment]
I think #3238 has somewhat little value until ``__array_function__`` is properly implemented in pint master. @keewis what's your opinion?

The other two only improve the tests so users of the release version don't benefit from them.

I'd say let's close it!

**Comment by keewis** [comment]
I agree that until `__array_function__` is at least in pint master (which may take a while, especially since hgrecco/pint#764 seems to be on hold), having #3238 in a release is not really important.

**Comment by jhamman** [comment]
Okay all. Making the release now.

**Comment by jhamman** [comment]
0.14 is on github/pypi/conda-forge/readthedocs now. Announcement emails/tweets are pending. Closing this now.

**Comment by dcherian** [comment]
:confetti_ball:

**Comment by max-sixty** [comment]
Thanks a lot @jhamman !
